File: youtubeapi/src/main/java/com/liskovsoft/youtubeapi/next/v1/WatchNextManagerSigned.java
package com.liskovsoft.youtubeapi.next.v1;

import com.liskovsoft.youtubeapi.next.v1.result.WatchNextResultContinuation;
import com.liskovsoft.youtubeapi.next.v1.result.WatchNextResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;

/**
 * For signed users!
 */
public interface WatchNextManagerSigned {
    @Headers("Content-Type: application/json")
    @POST("https://www.youtube.com/youtubei/v1/next")
    Call<WatchNextResult> getWatchNextResult(@Body String suggestQuery, @Header("Authorization") String auth);

    @Headers("Content-Type: application/json")
    @POST("https://www.youtube.com/youtubei/v1/next")
    Call<WatchNextResultContinuation> continueWatchNextResult(@Body String suggestQuery, @Header("Authorization") String auth);
}

File: youtubeapi/src/main/java/com/liskovsoft/youtubeapi/next/v2/WatchNextManagerSigned.kt
package com.liskovsoft.youtubeapi.next.v2

import com.liskovsoft.youtubeapi.next.v2.result.gen.WatchNextResult
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.Headers
import retrofit2.http.POST

/**
 * For signed users!
 */
interface WatchNextManagerSigned {
    @Headers("Content-Type: application/json")
    @POST("https://www.youtube.com/youtubei/v1/next")
    fun getWatchNextResult(@Body suggestQuery: String?, @Header("Authorization") auth: String?): Call<WatchNextResult?>?

    //@Headers("Content-Type: application/json")
    //@POST("https://www.youtube.com/youtubei/v1/next")
    //fun continueWatchNextResult(@Body suggestQuery: String?, @Header("Authorization") auth: String?): Call<WatchNextResultContinuation?>?
}